haystack/components/preprocessors/document_splitter.py

```diff
@@ -109,9 +111,15 @@ def run(self, documents: List[Document]):
         split_docs = []
         for doc in documents:
             if doc.content is None:
-                raise ValueError(
-                    f"DocumentSplitter only works with text documents but content for document ID {doc.id} is None."
+                logger.warning(
+                    "DocumentSplitter only works with text documents but content for document ID {doc_id} is None."
+                    " Skipping this document.",
+                    doc_id=doc.id,
```

Comment:
@davidsbatista let me know if you'd prefer this to be kept as a ValueError. In my opinion I'd rather let this fail forward, as in it skips Documents with content=None rather than fully stopping when it runs into this scenario.